python/paddle/fluid/dygraph/jit.py

```diff
@@ -166,6 +171,793 @@ def __impl__(*args, **kwargs):
+            fluid.dygraph.jit.save(
+                layer=net,
+                model_path=model_path,
+                input_spec=[img])
+    """
+
+    def get_inout_spec(all_vars, target_vars, return_name=False):
+        valid_vars = [var for var in all_vars if isinstance(var, Variable)]
+        valid_var_dict = {}
+        for var in valid_vars:
+            valid_var_dict[var.name] = var
+        if target_vars:
+            for i, var in enumerate(target_vars):
+                # check target var whether exists
+                if var.name not in valid_var_dict:
+                    raise RuntimeError(
+                        "The variable to feed/fetch are not exist.")
+                target_vars[i] = valid_var_dict[var.name]
+        else:
+            target_vars = valid_vars
+        if return_name:
+            target_vars = [var.name for var in target_vars]
+
+        return target_vars
+
+    # 1. input check
+    prog_translator = ProgramTranslator()
+    if not prog_translator.enable:
+        raise RuntimeError(
+            "The paddle.imperative.jit.save doesn't work when setting ProgramTranslator.enable=False."
+        )
+    if not isinstance(layer, Layer):
+        raise TypeError(
+            "The input layer of paddle.imperative.jit.save should be 'Layer', but received layer type is %s."
+            % type(layer))
+
+    if configs is None:
+        configs = SaveLoadConfig()
+
+    if input_spec is not None and not isinstance(input_spec, list):
+        raise TypeError(
+            "The input input_spec should be 'list', but received input_spec's type is %s."
+            % type(input_spec))
+        for var in input_spec:
+            if not isinstance(var, core.VarBase):
+                raise TypeError(
+                    "The element in input_spec list should be 'Variable', but received element's type is %s."
+                    % type(var))
+
+    # 2. get program of declarative Layer.forward
+    prog_cache = prog_translator.get_program_cache()
+    # make dummy args & kwargs, to get excepted FunctionSpec
+    layer_func = FunctionSpec(type(layer).forward, [layer], {})
+    concrete_program, _ = prog_cache.get_program(layer_func)
+
+    # 3. share parameters form Layer to scope & record var info
+    scope = core.Scope()
+    state_dict = layer.state_dict()
+    extra_var_info = dict()
+    for structured_name, param in state_dict.items():
+        # share to scope
+        param_tensor = scope.var(param.name).get_tensor()
+        src_tensor = param.value().get_tensor()
+        param_tensor._share_data_with(src_tensor)
+        # record var info
+        extra_info_dict = dict()
+        extra_info_dict['structured_name'] = structured_name
+        extra_info_dict['stop_gradient'] = param.stop_gradient
+        if isinstance(param, ParamBase):
+            extra_info_dict['trainable'] = param.trainable
+        extra_var_info[param.name] = extra_info_dict
```

Comment:
Do we need log the information of persistable buffers when isinstance(param, core.varBase)?

Reply:
here is no `trainable` info, we think it it a VarBase